network-config-analyzer/K8sPolicyYamlParser.py

```diff
@@ -153,18 +153,28 @@ def parse_port(self, port):
         :return: A ConnectionSet representing the allowed connections by this element (protocols X port numbers)
         :rtype: ConnectionSet
         """
-        self.check_keys_are_legal(port, 'port', {'port': 0, 'protocol': 0})
+        self.check_keys_are_legal(port, 'port', {'port': 0, 'endPort': 0, 'protocol': 0})
         port_id = port.get('port')
         protocol = port.get('protocol')
+        endport_id = port.get('endPort')
         if not protocol:
             protocol = 'TCP'
         if protocol not in ['TCP', 'UDP', 'SCTP']:
             self.syntax_error('protocol must be "TCP" or "UDP" or "SCTP"', port)
 
         res = ConnectionSet()
         dest_port_set = PortSet(port_id is None)
-        if port_id:
+        if port_id and endport_id:
+            if isinstance(port_id, str):
+                self.syntax_error("If Port is a string (named port) EndPort cannot be specified", port)
+            if port_id > endport_id:
```

Comment:
Should we give a warning if port_id == endport_id (in which case endPort is redundant)

Reply:
NO way!!!